ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/widgets/listtab/ListButtonWidget.java

```diff
@@ -29,106 +31,112 @@
 
 /**
  * @author Dmitry Shnurenko
+ * @author Vitaliy Guliy
  */
 public class ListButtonWidget extends Composite implements ListButton {
+
     interface ListButtonWidgetUiBinder extends UiBinder<Widget, ListButtonWidget> {
     }
 
-    private static final int Y_SHIFT = 21;
-
     private static final String GWT_POPUP_STANDARD_STYLE = "gwt-PopupPanel";
 
     private static final ListButtonWidgetUiBinder UI_BINDER = GWT.create(ListButtonWidgetUiBinder.class);
 
-    private final FlowPanel      listPanel;
     private final PopupPanel     popupPanel;
-    private final List<ListItem> items;
 
-    @UiField
-    FlowPanel button;
+    private final FlowPanel      listPanel;
+
+    private final List<ListItem> items = new ArrayList<>();
 
     @UiField(provided = true)
     final Resources resources;
 
     private ActionDelegate delegate;
-    private boolean        isShown;
+
+    private long closeTime;
 
     @Inject
-    public ListButtonWidget(Resources resources, FlowPanel listPanel, PopupPanel popupPanel) {
+    public ListButtonWidget(Resources resources) {
         this.resources = resources;
-
         initWidget(UI_BINDER.createAndBindUi(this));
 
-        this.listPanel = listPanel;
-        this.listPanel.addStyleName(resources.partStackCss().listItemPanel());
-
-        this.popupPanel = popupPanel;
-        this.popupPanel.removeStyleName(GWT_POPUP_STANDARD_STYLE);
-        this.popupPanel.add(listPanel);
-
-        addDomHandler(this, ClickEvent.getType());
-
-        this.items = new ArrayList<>();
+        closeTime = System.currentTimeMillis();
+        addDomHandler(new MouseDownHandler() {
+            @Override
+            public void onMouseDown(MouseDownEvent event) {
+                long time = System.currentTimeMillis();
+                if (time - closeTime < 100) {
```

Comment:
The popup panel is autoclosed and I had no idea how to prevent opening it after pressing menu button. So, decided to use time to detect the action.